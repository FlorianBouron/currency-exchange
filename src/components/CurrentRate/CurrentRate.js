import React from "react";
import PropTypes from "prop-types";
import TrendingUp from "@material-ui/icons/TrendingUp";
import styles from "./CurrentRate.module.scss";

const CurrentRate = ({ rateFrom, rateTo }) => {
  return (
    <div className={styles["current-rate-container"]}>
      <TrendingUp className={styles["current-rate-icon"]} />
      {rateFrom} = {rateTo}
    </div>
  );
};

CurrentRate.propTypes = {
  rateFrom: PropTypes.string.isRequired,
  rateTo: PropTypes.string.isRequired
};

export default CurrentRate;
