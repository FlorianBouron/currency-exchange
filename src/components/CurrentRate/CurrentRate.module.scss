@import "../../constants/utils.scss";
@import "../../constants/colors.scss";

.current-rate-container {
  padding: 0 px2rem(5px);
  border: solid px2rem(2px) $primary-gray;
  border-radius: px2rem(20px);
  display: inline-flex;
  align-items: center;
  color: $primary-blue;
  background-color: $white;

  .current-rate__icon {
    margin-right: px2rem(10px);
  }
}
