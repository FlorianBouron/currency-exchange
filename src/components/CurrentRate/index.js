import CurrentRate from "./CurrentRate";

export default CurrentRate;
