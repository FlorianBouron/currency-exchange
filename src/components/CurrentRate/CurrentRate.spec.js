import React from "React";
import { shallow } from "enzyme";
import CurrentRate from ".";

describe("CurrentRate component", () => {
  const component = shallow(<CurrentRate rateFrom="1zl" rateTo="0,231â‚¬" />);
  it("has working component", () => {
    expect(component).toBeDefined();
  });
  it("expects component to never be changed", () => {
    expect(component).toMatchSnapshot();
  });
});
