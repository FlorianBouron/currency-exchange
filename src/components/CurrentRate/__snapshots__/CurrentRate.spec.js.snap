// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentRate component expects component to never be changed 1`] = `
<div
  className="current-rate-container"
>
  <[object Object]
    className="current-rate-icon"
  />
  1zl
   = 
  0,231â‚¬
</div>
`;
